example_addressgroup_test.go

```diff
@@ -0,0 +1,135 @@
+	addr1 := addr.Entry{
+		Name:        "SampleAddress1",
+		Value:       "10.192.226.101/32",
+		Type:        addr.IpNetmask,
+		Description: "First address of a sample address group",
+		Tags:        tags,
+	}
+	if err = pan.Objects.Address.Set(deviceGroup, addr1); err != nil {
+		log.Panic(err)
+	}
+
+	addr2 := addr.Entry{
+		Name:        "SampleAddress2",
+		Value:       "10.192.226.102/32",
+		Type:        addr.IpNetmask,
+		Description: "Second address of a sample address group",
+		Tags:        tags,
+	}
+	if err = pan.Objects.Address.Set(deviceGroup, addr2); err != nil {
+		log.Panic(err)
+	}
+
+	ag := addrgrp.Entry{
+		Name:            "SampleAddressGroup",
+		Description:     "This in an example on how to use address groups",
+		StaticAddresses: []string{addr1.Name, addr2.Name},
+		Tags:            tags,
+	}
+	if err = pan.Objects.AddressGroup.Set(deviceGroup, ag); err != nil {
+		log.Panic(err)
+	}
+
+	securityPolicy := security.Entry{
+		Name:                 "SamplePolicy",
+		Description:          "This is where the request number goes",
+		Tags:                 tags,
+		SourceZones:          []string{"CORPEXT"},
+		SourceAddresses:      []string{"any"},
+		DestinationZones:     []string{"CORPDMZ"},
+		DestinationAddresses: []string{ag.Name},
+		Applications:         []string{"ssl"},
+		Services:             []string{"application-default"},
+		LogSetting:           "Standard-Logging",
+		Group:                "Corp_Default",
+	}
+	securityPolicy.Defaults()
+
+	if err = pan.Policies.Security.VerifiableSet(deviceGroup, util.PreRulebase, securityPolicy); err != nil {
+		log.Panic(err)
+	}
+
+	panCommit := commit.PanoramaCommit{
+		Description:  "Created example address group",
+		Admins:       nil,
+		DeviceGroups: []string{deviceGroup},
+	}
+
+	resp, bytes, err := pan.Commit(panCommit, "", nil)
+	if err != nil {
+		log.Panic(err)
+	}
+	log.Printf("Job ID: %v\n", resp)
+	log.Printf("Response XML: %v\n", string(bytes))
+
+	// Delete the addresses, address group and security policy
+	securityPolicy, err = pan.Policies.Security.Get(deviceGroup, util.PreRulebase, "SamplePolicy")
+	if err != nil {
+		log.Panic(err)
+	}
+	if err = pan.Policies.Security.Delete(deviceGroup, util.PreRulebase, securityPolicy); err != nil {
+		log.Panic(err)
+	}
+
+	for _, agn := range securityPolicy.DestinationAddresses {
+		ag, err := pan.Objects.AddressGroup.Get(deviceGroup, agn)
+		if err != nil {
+			log.Panic(err)
+		}
+		if err = pan.Objects.AddressGroup.Delete(deviceGroup, ag); err != nil {
+			log.Panic(err)
+		}
+		for _, an := range ag.StaticAddresses {
+			a, err := pan.Objects.Address.Get(deviceGroup, an)
+			if err = pan.Objects.Address.Delete(deviceGroup, a); err != nil {
+				log.Panic(err)
+			}
+		}
+	}
+
+	panCommit = commit.PanoramaCommit{
+		Description:  "Deleted sample address group",
+		Admins:       nil,
+		DeviceGroups: []string{deviceGroup},
+	}
+
+	resp, bytes, err = pan.Commit(panCommit, "", nil)
```

Comment:
So you commit on line # 90 but you don't wait for that commit to finish before this line is invoked.  I know Panorama will let you run multiple commits, but I'm not sure if it'll let you run multiple commits against the same device group.  Can you confirm if this code works against a live PAN-OS without the need to wait for the commit to finish?

Reply:
I was actually wondering about this.  It seems to work fine, but I agree it would be better to let it finish.  What happened to the sync parameter that was in v0.4.0?  It had this:
// Param sync should be true if you want this function to block until the commit job completes.

Reply:
Yeah, I removed the sync param.  Panorama allows users to submit multiple commits at once, and when I moved the code to normalized commits, I felt like it wasn't `Commit()`s responsibility to wait for the job to complete (as there are repeated function calls to wait for completion), plus it muddies the waters of what a failure means when you ask for a commit.

Since there's already a `WaitForJob()` I just leave it up to the user to wait for the job if they want.

Reply:
Gotcha, hadn't noticed that function.

Error handling is a pretty big deal and one the areas where using the raw api has an edge over Terraform, specially in a world that includes a mix of manual changes and automated ones.  Will try to figure out how to get that working and I'll add it in there in another PR

Thanks again Garfield